app/src/main/java/org/mifos/mobile/ui/home/HomeScreen.kt

```diff
@@ -0,0 +1,303 @@
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.mifos.mobile.R
+import org.mifos.mobile.core.ui.component.MifosHiddenTextRow
+import org.mifos.mobile.core.ui.component.MifosLinkText
+import org.mifos.mobile.core.ui.component.MifosUserImage
+import org.mifos.mobile.core.ui.theme.MifosMobileTheme
+import org.mifos.mobile.utils.CurrencyUtil
+
+
+@Composable
+fun HomeScreen(
+    username: String,
+    totalLoanAmount: Double,
+    totalSavingsAmount: Double,
+    userBitmap: Bitmap?,
+    userProfile: () -> Unit,
+    totalSavings: () -> Unit,
+    totalLoan: () -> Unit,
+    callHelpline: (String) -> Unit,
+    mailHelpline: (String) -> Unit,
+    homeCardClicked: (HomeCardItem) -> Unit
+) {
+    val scrollState = rememberScrollState()
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(horizontal = 16.dp)
+            .verticalScroll(scrollState)
+    ) {
+        UserDetailsRow(
+            userBitmap = userBitmap,
+            username = username,
+            userProfile = userProfile
+        )
+
+        Spacer(modifier = Modifier.height(8.dp))
+
+        AccountOverviewCard(
+            totalLoanAmount = totalLoanAmount,
+            totalSavingsAmount = totalSavingsAmount,
+            totalLoan = totalLoan,
+            totalSavings = totalSavings
+        )
+
+        Spacer(modifier = Modifier.height(8.dp))
+
+        HomeCards(homeCardClicked = homeCardClicked)
+
+        ContactUsRow(callHelpline = callHelpline, mailHelpline = mailHelpline)
+    }
+}
+
+@Composable
+private fun HomeCards(
+    homeCardClicked: (HomeCardItem) -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+    ) {
+        val items = getHomeCardItems()
+
+        for (i in items.indices step 3) {
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(bottom = 8.dp),
+                horizontalArrangement = Arrangement.spacedBy(8.dp)
+            ) {
+                for (j in 0 until minOf(3, items.size - i)) {
+                    val index = i + j
+                    HomeCard(
+                        modifier = Modifier.weight(1f),
+                        titleId = items[index].titleId,
+                        drawableResId = items[index].drawableResId,
+                        onClick = { homeCardClicked(items[index]) }
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun UserDetailsRow(
+    userBitmap: Bitmap?,
+    username: String,
+    userProfile: () -> Unit
+) {
+    val interactionSource = remember { MutableInteractionSource() }
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(20.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        userBitmap?.let {
```

Comment:
if user bitmap is null then show first char of user name in image circular holder